src/processor/blockchain-events/agreement.ts

```diff
@@ -66,7 +70,10 @@ const handlers: HandlersObject<BlockchainAgreementEvents, BlockchainEventProcess
     agreement.isActive = false
     await agreement.save()
 
-    if (options.manager) await options.manager.unpin(agreement.dataReference)
+    if (options.manager) {
+      await options.manager.unpin(agreement.dataReference)
+      channel.broadcast(MessageCodesEnum.I_AGREEMENT_STOPPED, { agreementReference: agreement.agreementReference })
```

Comment:
Could we provide a separate class that will handle pinning with jobs, something like ProvideWithJobsManager?
Not sure how much effort it will give us)

Reply:
Not sure what you mean 😳 Can you please clarify?

Reply:
I mean at some point when we have more logic in our event processing we should think about splitting this part? 
Is the ProviderManager somehow related to Communication?

Reply:
Well this is about Agreements not Pinning Jobs. It is in the `if` because we don't have the channel when precaching.

I agree it is not completely great solution but to me I don't really see major benefit in looking for some other solution atm.